docs/topics/collection-elements.md

```diff
@@ -123,6 +123,29 @@ fun main() {
 ```
 {kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
 
+## Retrieve with selector
+
+If you need to map the collection before retrieving the element, there is a function [`firstNotNullOf()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-not-null-of.html).
+It combines 2 actions:
+- Maps the collection with the selector function
+- Returns the first non-null value in the result
+
+`firstNotNullOf()` throws the `NoSuchElementException` if the resulting collection doesn't have a non-null element. 
+Use the counterpart [`firstNotNullOfOrNull()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/first-not-null-of-or-null.html) 
+to return null in this case.
+
+```kotlin
+fun main() {
+//sampleStart
+   val list = listOf("one", "two", "three")
+   // Returns the first string that has required length
+   val longEnough = list.firstNotNullOf { it.takeIf { it.length > 4 } }
```

Comment:
This way it can be replaced with just a plain 'first` with predicate because no mapping is done here. Let's preserve the mapping in the example.

Reply:
Oh, I see. Thank you, will fix.

Reply:
I preserved the initial example but gave a name to an outer implicit `it` not to produce shadowing